### ISSUE TITLE
Skygear migrate required root to perform migration on database

### ISSUE THREAD

**Issue by rickmak** [SkygearIO/skygear-server#1136]
Skygear-version: quay.io/skygeario/skygear-migrate:git-00a3b0f 

Command involved.
```
docker run --rm \
		quay.io/skygeario/skygear-migrate:git-00a3b0f \
		-add-migration-src=core,github://:@skygeario/skygear-server/migrations/core#00a3b0f \
		-add-migration-src=auth,github://:@skygeario/skygear-server/migrations/auth#00a3b0f \
		-migration=core \
		-migration=auth \
		-schema=app_myapp \
		-database=[DBURL] \
		up
```
#### Steps to reproduce

- Create an user used with grant full access right to the DB with follow command.
```
CREATE USER "skygear"
    WITH ENCRYPTED PASSWORD '';
GRANT ALL PRIVILEGES ON DATABASE skygear
    TO "skygear";
```
- Run the above command with DBURL contains db user with only DB level permission.
- Follow error will print on the console

```
time="2019-11-27T14:59:10Z" level=error msg="pq: no schema has been selected to create in in line 0: CREATE TABLE \"_core_version\" (version bigint not null primary key, dirty boolean not null)" db_host=[HOST] db_name=/skygear error="pq: no schema has been selected to create in in line 0: CREATE TABLE \"_core_version\" (version bigint not null primary key, dirty boolean not null)" migration=core schema=app_leighton source_url="file:///tmp/skygear-migrate-src/skygeario-skygear-server-00a3b0f/SkygearIO-skygear-server-00a3b0f/migrations/core"
make: *** [skygear-migrate] Error 1
```

If I replace the DBURL with root connection string, it will migrate successfully with following output:

```
time="2019-11-27T15:05:19Z" level=info msg=Done db_host=[HOST] db_name=/skygear migration=core result=ok schema=app_leighton source_url="file:///tmp/skygear-migrate-src/skygeario-skygear-server-00a3b0f/SkygearIO-skygear-server-00a3b0f/migrations/core"
time="2019-11-27T15:05:21Z" level=info msg=Done db_host=[HOST] db_name=/skygear migration=auth result=ok schema=app_leighton source_url="file:///tmp/skygear-migrate-src/skygeario-skygear-server-00a3b0f/SkygearIO-skygear-server-00a3b0f/migrations/auth"
```

The DB env is Azure Database for PostgreSQL Single server with SKU=GP_Gen5_2

**Comment by rickmak** [comment]
Tried to grant the permission by following for deployment user but no luck

```
GRANT ALL ON SCHEMA app_myapp to "skygear";
```

**Comment by rickmak** [comment]
I found that if I DROP DATABASE and start over with exact following, it will works


```
CREATE DATABASE skygear;
CREATE USER "skygear"
    WITH ENCRYPTED PASSWORD '';
GRANT ALL PRIVILEGES ON DATABASE skygear
    TO "skygear";
CREATE SCHEMA IF NOT EXISTS app_myapp;¬
GRANT ALL ON SCHEMA app_myapp to "skygear";
```

In the previous report, I forgot to run `GRANT ALL ON SCHEMA app_myapp to "skygear";` and use the root account to run the migration once. After the migration, I grant the schema right but it is too late?

Not exactly a skygear issue but postgresql permission. Closing.
